includes/db-adapter.php

```diff
@@ -93,6 +105,69 @@ private function get_suggested_replacements( $entry ) {
 		return $suggestions;
 	}
 
+	private function morphology_matches( $entry, $suggestion ) {
+		$entry_morphology = $this->rosette_api_morphology( $entry->singular );
+		$suggestion_morphology = $this->rosette_api_morphology( $suggestion->singular );
+
+		if ( ! $entry_morphology || ! $suggestion_morphology ) {
+			return true;
+		}
+
+		$entry_lemmas_count = array_count_values( $entry_morphology->lemmas );
+		$suggestion_lemmas_count = array_count_values( $suggestion_morphology->lemmas );
+
+		foreach ( $entry_morphology->lemmas as $entry_key => $lemma ) {
+			// Can't easily compare lemmas that exist more than once in one of the strings.
+			if ( $entry_lemmas_count[ $lemma ] > 1 || $suggestion_lemmas_count[ $lemma ] > 1 ) {
+				continue;
+			}
+
+			// Does that lemma also exist in our suggestion?
+			if ( false !== $suggestion_key = array_search( $lemma, $suggestion_morphology->lemmas, true ) ) {
+				// Is it the same part of speech?
+				if ( $suggestion_morphology->posTags[ $suggestion_key ] !== $entry_morphology->posTags[ $entry_key ] ) {
```

Comment:
Actually not necessary, since we start with the lemma - so if we have a punctuation (ex: `.` ) in both strings, the POS will always be the same.